crates/holochain_state/src/source_chain.rs

```diff
@@ -1609,40 +1617,31 @@ pub mod tests {
         .await
         .unwrap();
 
-        {
-            let chain = SourceChain::new(
-                db.clone(),
-                dht_db.to_db(),
-                dht_db_cache.clone(),
-                keystore.clone(),
-                alice.clone(),
-            )
-            .await?;
-            assert_eq!(
-                chain
-                    .valid_cap_grant(function.clone(), alice.clone(), secret.clone())
-                    .await?,
-                Some(CapGrant::ChainAuthor(alice.clone())),
-            );
+        let chain = SourceChain::new(
+            db.clone(),
+            dht_db.to_db(),
+            dht_db_cache.clone(),
+            keystore.clone(),
+            alice.clone(),
+        )
+        .await?;
+        assert_eq!(
+            chain
+                .valid_cap_grant(function.clone(), alice.clone(), secret.clone())
+                .await?,
+            Some(CapGrant::ChainAuthor(alice.clone())),
+        );
```

Comment:
Hmm how is it that a grant with no secret is returned, even though a secret is provided? Because shouldn't that pass through [this filter of the query](https://github.com/holochain/holochain/pull/2097/files#diff-13ceb40c45d19a10c53bf8f5e7a4319bdd8f2d1fcc2fb4072beebfd245981816R647), requiring the cap to have the secret provided?

Reply:
[If the caller is the chain owner](https://github.com/holochain/holochain/pull/2097/files#diff-13ceb40c45d19a10c53bf8f5e7a4319bdd8f2d1fcc2fb4072beebfd245981816R630), it's an author cap grant and any further check is bypassed. I'll add a comment for that.